File: ios/telobike/telobike/TBCity.m
//
//  TBCity.m
//  telobike
//
//  Created by Elad Ben-Israel on 9/27/13.
//  Copyright (c) 2013 Elad Ben-Israel. All rights reserved.
//

#import "TBCity.h"
#import "Utils.h"

@interface TBCity ()

@property (strong, nonatomic) NSDictionary* dict;

@end

@implementation TBCity

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    if (self) {
        self.dict = dict;
    }
    return self;
}

- (NSString*)cityName { return [[_dict objectForKey:@"city_name.en"] stringByReplacingOccurrencesOfString:@"-" withString:@" "]; }
- (NSString*)mail { return [_dict objectForKey:@"mail"]; }
- (NSString*)serviceName { return [_dict localizedStringForKey:@"service_name"]; }
- (NSString*)mailTags { return [_dict objectForKey:@"mail_tags"]; }
- (NSArray*)messages { return [_dict objectForKey:@"messages"]; }
- (CLLocation*)cityCenter { return [_dict locationForKey:@"city_center"]; }
- (NSString*)disclaimer { return [_dict objectForKey:@"disclaimer"]; }
- (NSURL*)infoURL { return [_dict urlForKey:@"info_url"]; }

- (CLCircularRegion *)region {
    return [[CLCircularRegion alloc] initWithCenter:CLLocationCoordinate2DMake(32.08282450, 34.77996850) radius:9108.0f identifier:self.cityName];
}

@end

File: ios/telobike/telobike/TBCity.swift
//
//  City.swift
//  telobike
//
//  Created by Elad Ben-Israel on 9/18/14.
//  Copyright (c) 2014 Elad Ben-Israel. All rights reserved.
//

import Foundation

class TBCity: NSObject {
    private let dict: NSDictionary
    
    init(dictionary: NSDictionary) {
        self.dict = dictionary
    }
    
    var cityName: String {
        return self.dict.stringForKey("city_name.en").stringByReplacingOccurrencesOfString("-", withString: " ")
    }
    
    var mail: String {
        return self.dict.stringForKey("mail")
    }
    
    var serviceName: String? {
        return self.dict.localizedStringForKey("service_name")
    }
    
    var mailTags: String {
        return self.dict.stringForKey("mail_tags")
    }
    
    var cityCenter: CLLocation {
        return self.dict.locationForKey("city_center")
    }
    
    var disclaimer: String? {
        return self.dict.stringForKey("disclaimer")
    }
    
    var infoURL: NSURL? {
        return self.dict.urlForKey("info_url")
    }
    
    var region: CLCircularRegion? {
        return CLCircularRegion(
            center: self.cityCenter.coordinate,
            radius: 9108.0,
            identifier: self.cityName)
    }
}